resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodsSnapshot.scala

```diff
@@ -65,14 +72,35 @@ object ExecutorPodsSnapshot extends Logging {
         case "pending" =>
           PodPending(pod)
         case "running" =>
+          // If we're checking all containers look for any non-zero exits
           if (shouldCheckAllContainers &&
             "Never" == pod.getSpec.getRestartPolicy &&
             pod.getStatus.getContainerStatuses.stream
               .map[ContainerStateTerminated](cs => cs.getState.getTerminated)
               .anyMatch(t => t != null && t.getExitCode != 0)) {
             PodFailed(pod)
           } else {
-            PodRunning(pod)
+            // Otherwise look for the Spark container
+            val sparkContainerStatusOpt = pod.getStatus.getContainerStatuses.asScala
+              .find(_.getName() == sparkContainerName)
+            sparkContainerStatusOpt match {
+              case Some(sparkContainerStatus) =>
+                sparkContainerStatus.getState.getTerminated match {
+                  case t if t.getExitCode != 0 =>
+                    PodFailed(pod)
+                  case t if t.getExitCode == 0 =>
+                    PodSucceeded(pod)
+                  case _ =>
+                    logWarning(s"Unable to find container ${sparkContainerName} in pod ${pod} " +
```

Comment:
Shall we revise this because we already find a corresponding container in this code path via `Some(sparkContainerStatus)` while we don't know the termination code?